go/ql/lib/semmle/go/frameworks/Fasthttp.qll

```diff
@@ -0,0 +1,452 @@
+ * from the `github.com/valyala/fasthttp` package.
+ */
+
+import go
+private import semmle.go.security.RequestForgeryCustomizations
+
+/**
+ * Provides classes for working with the [fasthttp](github.com/valyala/fasthttp) package.
+ */
+module Fasthttp {
+  /** Gets the v1 module path `github.com/valyala/fasthttp`. */
+  string v1modulePath() { result = "github.com/valyala/fasthttp" }
+
+  /** Gets the path for the root package of fasthttp. */
+  string packagePath() { result = package(v1modulePath(), "") }
+
+  private class HeaderWrite extends Http::HeaderWrite::Range, DataFlow::MethodCallNode {
+    string methodName;
+
+    HeaderWrite() {
+      this.getTarget().hasQualifiedName(packagePath(), "ResponseHeader", methodName) and
+      methodName in [
+          "Add", "AddBytesK", "AddBytesKV", "AddBytesV", "Set", "SetBytesK", "SetBytesKV",
+          "SetCanonical", "SetContentType", "SetContentTypeBytes"
+        ]
+      or
+      this.getTarget().hasQualifiedName(packagePath(), "RequestCtx", methodName) and
+      methodName in ["SetContentType", "SetContentTypeBytes", "Success", "SuccessString"]
+    }
+
+    override DataFlow::Node getName() {
+      methodName =
+        [
+          "Add", "AddBytesK", "AddBytesKV", "AddBytesV", "Set", "SetBytesK", "SetBytesKV",
+          "SetCanonical"
+        ] and
+      result = this.getArgument(0)
+    }
+
+    override string getHeaderName() {
+      result = Http::HeaderWrite::Range.super.getHeaderName()
+      or
+      methodName = ["SetContentType", "SetContentTypeBytes", "Success", "SuccessString"] and
+      result = "content-type"
+    }
+
+    override DataFlow::Node getValue() {
+      if methodName = ["SetContentType", "SetContentTypeBytes", "Success", "SuccessString"]
+      then result = this.getArgument(0)
+      else result = this.getArgument(1)
+    }
+
+    override Http::ResponseWriter getResponseWriter() {
+      result.(ResponseWriter).getAHeaderObject() = this
+    }
+  }
+
+  private class ResponseWriter extends Http::ResponseWriter::Range {
+    SsaWithFields v;
+
+    ResponseWriter() {
+      this = v.getBaseVariable().getSourceVariable() and
+      (
+        v.getType().hasQualifiedName(packagePath(), ["Response", "ResponseHeader"]) or
+        v.getType().(PointerType).getBaseType().hasQualifiedName(packagePath(), "RequestCtx")
+      )
+    }
+
+    override DataFlow::Node getANode() { result = v.similar().getAUse().getASuccessor*() }
+
+    /** Gets a header object that corresponds to this HTTP response. */
+    DataFlow::MethodCallNode getAHeaderObject() {
+      result.getTarget().getName() =
+        [
+          "Add", "AddBytesK", "AddBytesKV", "AddBytesV", "Set", "SetBytesK", "SetBytesKV",
+          "SetCanonical", "SetContentType", "SetContentTypeBytes", "Success", "SuccessString"
+        ] and
+      this.getANode() = result.getReceiver()
+    }
+  }
+
+  private class ResponseBody extends Http::ResponseBody::Range {
+    DataFlow::MethodCallNode call;
```

Comment:
Please give `call` a better name, which describes what it does. Better yet, replace this field with `DataFlow::Node responseWriter`, set it to the response writer in the first place and return it in `getResponseWriter`.